src/tools/tidy/src/deps.rs

```diff
@@ -515,6 +517,19 @@ pub fn check(root: &Path, cargo: &Path, bad: &mut bool) {
     let mut checked_runtime_licenses = false;
 
     for &(workspace, exceptions, permitted_deps) in WORKSPACES {
+        let gitmodules = root.join(".gitmodules");
```

Comment:
Can we move this into the function? It seems like the function could take the directory rather than the file path, which simplifies it and avoids duplicating the exact filename everywhere.